src/jvm/main/org/jetbrains/kotlinx/lincheck/Reporter.kt

```diff
@@ -80,28 +80,53 @@ private class ActorWithResult(val actorRepresentation: String, val spacesAfterAc
                                     " ".repeat(spacesAfterResult) + clockRepresentation
 }
 
-private fun uniteActorsAndResultsLinear(actors: List<Actor>, results: List<Result>): List<ActorWithResult> {
+private fun uniteActorsAndResultsLinear(
+    actors: List<Actor>,
+    results: List<Result>,
+    exceptionStackTraces: Map<Throwable, ExceptionNumberAndStacktrace>
+): List<ActorWithResult> {
     require(actors.size == results.size) {
         "Different numbers of actors and matching results found (${actors.size} != ${results.size})"
     }
     return actors.indices.map {
-        ActorWithResult("${actors[it]}", 1, "${results[it]}", 0, "")
+        val result = results[it]
+        ActorWithResult(
+            actorRepresentation = "${actors[it]}",
+            spacesAfterActor = 1,
```

Comment:
wouldn't it be better to specify the total length instead? Now, you have an invariant that `actorRepresentation.size + spacesAfterActor == resultRepresentation.size + spacesAfterResult`, which is easy to break accidentally.

Reply:
I suggest to make refactoring in a separate ticket.

Reply:
Sure, let's refactor Reporter and TraceReporter files, but let's also not introduce new design mistakes.

Reply:
No, here I don't have such constrains (`actorRepresentation.size + spacesAfterActor == resultRepresentation.size + spacesAfterResult`) this method is called only to constuct output for Init and Post parts.